client/my-sites/themes/theme-options.js

```diff
@@ -66,10 +91,10 @@ const activate = {
 	action: activateAction,
 	hideForTheme: ( state, themeId, siteId ) => (
 		! getCurrentUser( state ) ||
-		( isJetpackSite( state, siteId ) && isJetpackSiteMultiSite( state, siteId ) ) ||
+		isJetpackSiteMultiSite( state, siteId ) ||
 		isThemeActive( state, themeId, siteId ) ||
 		( isThemePremium( state, themeId ) && ! isPremiumThemeAvailable( state, themeId, siteId ) ) ||
-		( isJetpackSite( state, siteId ) && ! isThemeAvailableOnJetpackSite( state, themeId, siteId ) )
+		! isThemeAvailableOnJetpackSite( state, themeId, siteId )
```

Comment:
Again, with the same reasoning than above, it's enough with this selector because it calls `isJetpackSite` on its own. Otherwise we'll be calling it once here and again inside `hasJetpackSiteJetpackThemesExtendedFeatures`.